### ISSUE TITLE
EC2 AWS: Cloudformation

### ISSUE THREAD

**Issue by defunctio** [trailofbits/algo#132]
In a project to implement Algo as a native mobile application to both iOS and Android it was necessary to migrate much of the AWS infrastructure deployment for Algo into a [Cloudformation](https://aws.amazon.com/cloudformation/) template to eliminate dependencies, simplify development and reduce the required infrastructure to deploy Algo as a mobile application.

There are a few benefits over using Cloudformation on AWS rather than programmatically managing the resources necessary for Algo as we are currently doing.

- All resources are created, owned and managed by the stack owner.
- All resources are unique named to prevent naming collisions and promote a multi-user environment.
- Better resource limit management and cost estimation.
- Destroying a stack removes ALL resources associated with that instance of Algo.

In short, it simplifies managing all the necessary resources for the end user and prevents mistakes. It may be worth looking into the plausibility of reusing the template for Algo itself in the future.
